File: frogment/src/main/java/ch/grze/frogment/activity/ActivityComponentInjector.java
package ch.grze.frogment.activity;

import android.app.Activity;

import ch.grze.frogment.core.Core;
import ch.grze.frogment.core.component.AbstractActivityComponentInjector;

public class ActivityComponentInjector extends AbstractActivityComponentInjector {
    @Override
    public void inject(Core core, Activity activity) {
        try {
            final FrogmentActivityInterface frogmentActivity = getTypedActivity(activity);

            final FrogmentActivityComponent component = new FrogmentActivityComponent(core, frogmentActivity);
            frogmentActivity.setFrogmentActivityComponent(component);
        } catch (ClassCastException ignored) {}

        try {
            final StateAwareFrogmentActivityInterface stateAwareFrogmentActivity = getTypedActivity(activity);

            final StateAwareFrogmentActivityComponent component = new StateAwareFrogmentActivityComponent(core, stateAwareFrogmentActivity);
            stateAwareFrogmentActivity.setStateAwareFrogmentActivityComponent(component);
        } catch (ClassCastException ignored) {}
    }
}

File: frogment/src/main/java/ch/grze/frogment/activity/ActivityComponentInjector.kt
package ch.grze.frogment.activity

import android.app.Activity
import ch.grze.frogment.core.Core
import ch.grze.frogment.core.component.ComponentInjector

class ActivityComponentInjector : ComponentInjector<Activity> {
    override fun inject(core: Core, activity: Activity) {
        (activity as? FrogmentActivityInterface)
            ?.frogmentActivityComponent
            ?.core = core

        (activity as? StateAwareFrogmentActivityInterface<*>)
            ?.stateAwareFrogmentActivityComponent
            ?.core = core
    }
}